### DISCUSSION TITLE
getting "Provider routines:rsa_setup_md:digest not allowed:" error while generating certificate request openssl3.2.0 with FIPS provider

### DISCUSSION THREAD

**Discussion by reguripradeep** [openssl/openssl#23714] [Q&A] [answered]
I am trying to generate a certificate request with the following command.
openssl req -newkey rsa:2048 -sha1 -passout stdin -out apachesslcertrequest.pem -config spsopensslconfig.conf (with only fips provider)
This is working fine in openssl 1.0.2z series. But failing with openssl3.2.0
i see the following log:
OpenSSL output: EC080000:error:1C8000AE:Provider routines:rsa_setup_md:digest not allowed:providers\implementations\signature\rsa_sig.c:304:digest=sha1
 please help if there is any thing missed or if it is not supported? i dont see anything in the documentation

**Comment by t8m** [comment, marked as answer]
SHA1 is not a fips approved algorithm. Do not use the `-sha1` option.

**Reply by reguripradeep** [reply]
Got it but i am confused because it was working finein 1.0.2.z versons.
